AutoCollection/ServerRequestParser.ts

```diff
@@ -169,6 +169,29 @@ class ServerRequestParser extends RequestParser {
         return value;
     }
 
+    private parseHeaders(request:http.ServerRequest, parentId?: string) {
+        this.rawHeaders = request.headers || (<any>request).rawHeaders;
+        this.userAgent = request.headers && request.headers["user-agent"];
+        this.sourceCorrelationId = Util.getCorrelationContextTarget(request, RequestResponseHeaders.requestContextSourceKey);
+
+        if (request.headers) {
+            this.correlationContext = request.headers[RequestResponseHeaders.correlationContextHeader];
+
+            if (request.headers[RequestResponseHeaders.requestIdHeader]) {
+                this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
+                this.requestId = CorrelationIdManager.generateRequestId(this.parentId);
```

Comment:
The manually passed in parent id was to allow filling requestId from `correlationContext.operation.parentId` when manually calling trackRequestSync. TrackRequestSync is too late to get the value from headers (since it should be the last call in the request handler). If you do, you won't be correlated with any of the dependency telemetry that used a different parentId.